library/java/test/io/envoyproxy/envoymobile/engine/JvmBridgeUtilityTest.kt

```diff
@@ -0,0 +1,58 @@
+package io.envoyproxy.envoymobile.engine
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class JvmBridgeUtilityTest {
+
+  @Test
+  fun `retrieveHeaders produces a Map with all headers provided via passHeaders`() {
+    val utility = JvmBridgeUtility()
+    utility.passHeader("test-0".toByteArray(), "value-0".toByteArray(), true)
+    utility.passHeader("test-1".toByteArray(), "value-1".toByteArray(), false)
+    utility.passHeader("test-1".toByteArray(), "value-2".toByteArray(), false)
+
+    val headers = utility.retrieveHeaders()
+    val expectedHeaders = mapOf(
+      "test-0" to listOf("value-0"),
+      "test-1" to listOf("value-1", "value-2")
+    )
+
+    assertThat(headers)
+      .hasSize(2) // Two keys / header name
+      .usingRecursiveComparison().isEqualTo(expectedHeaders)
+  }
+
+  @Test
+  fun `validateCount checks if the expected number of header values in the map matches the actual`() {
```

Comment:
I think we can have 2 separate tests (potentially 3 but up to you):
1. `validateCount` returns the appropriate value (this can be split into the positive and negative cases but up to you)
2. `retrieveHeaders()` resets the internal count

Separate question: Wouldn't it be easier if the utility just returned the count value instead of requiring the user to ask it if the utility has N counts?

Reply:
We could do that. I don't know that it would necessarily be easier, but it's definitely a valid approach. I was trying to keep the interface scoped pretty narrowly, and nothing really has any business querying the accumulated count outside of validation purposes.

Reply:
Oh also, I'm fine with splitting up the tests.

Reply:
Let's split them up :)

I think it'll be clearer